### DISCUSSION TITLE
Support CycloneDX reachability

### DISCUSSION THREAD

**Discussion by schuman0** [aquasecurity/trivy#5616] [Ideas]
### Description

With release 9.9.0 cdxgen now supports reachability in their SBOMs: https://github.com/CycloneDX/cdxgen/releases/tag/v9.9.0

It would be awesome if Trivy could add this to the SBOMs it generates (and to the vuln reports)

### Target

None

### Scanner

Vulnerability

**Comment by DmitriyLewen** [comment]
Hello @schuman0 
Thanks for your interest to Tivy!

I'm not sure if this is possible currently.
Trivy scans (for vulnerabilities) only files with dependencies (`pom.xml`, `package-lock.json`, `package.json`, etc.).
We do not parse language code files (I mean `*.js`, `*.class`, etc.).

Correct me if I misunderstood you.

Regards, Dmitriy

**Comment by schuman0** [comment]
Can you integrate with their envise tool to provide this? https://cyclonedx.github.io/cdxgen/#/ADVANCED?id=evinse-mode-saasbom

**Reply by knqyf263** [reply]
You can create a plugin for that.
https://aquasecurity.github.io/trivy/v0.47/docs/advanced/plugins/
